modVM/src/org/aion/vm/ExecutionHelper.java

```diff
@@ -63,126 +69,78 @@ public Call(byte[] data, byte[] destination, byte[] value) {
         deleteAccounts.add(address);
     }
 
     /**
-     * Adds a collections of deleted accounts.
+     * Adds the collection addresses to the set of deleted accounts if addresses is non-null.
      *
-     * @param addresses
+     * @param addresses The addressed to add to the set of deleted accounts.
      */
     public void addDeleteAccounts(Collection<Address> addresses) {
-        for (Address address : addresses) {
-            addDeleteAccount(address);
+        for (Address addr : addresses) {
+            if (addr != null) { deleteAccounts.add(addr); }
         }
     }
 
     /**
-     * Returns logs.
+     * Adds log to the execution logs.
      *
-     * @return
-     */
-    public List<Log> getLogs() {
-        return logs;
-    }
-
-    /**
-     * Adds a log.
-     *
-     * @param log
+     * @param log The log to add to the execution logs.
      */
     public void addLog(Log log) {
         logs.add(log);
     }
 
     /**
-     * Adds a collection of logs.
+     * Adds a collection of logs, logs, to the execution logs.
      *
-     * @param logs
+     * @param logs The collection of logs to add to the execution logs.
      */
     public void addLogs(Collection<Log> logs) {
-        this.logs.addAll(logs);
-    }
-
-    /**
-     * Returns calls.
-     *
-     * @return
-     */
-    public List<Call> getCalls() {
-        return calls;
+        for (Log log : logs) {
+            if (log != null) { this.logs.add(log); }
+        }
     }
 
     /**
-     * Adds a call.
+     * Adds a call whose parameters are given by the parameters to this method.
      *
-     * @param data
-     * @param destination
-     * @param value
+     * @param data The call data.
+     * @param destination The call destination.
+     * @param value The call value.
      */
     public void addCall(byte[] data, byte[] destination, byte[] value) {
         calls.add(new Call(data, destination, value));
     }
 
     /**
-     * Returns internal transactions.
-     *
-     * @return
-     */
-    public List<AionInternalTx> getInternalTransactions() {
-        return internalTxs;
-    }
-
-    /**
-     * Adds an internal transaction.
+     * Adds an internal transaction, tx, to the internal transactions list.
      *
-     * @param tx
+     * @param tx The internal transaction to add.
      */
-    public void addInternalTransaction(AionInternalTx tx) {
-        internalTxs.add(tx);
-    }
+    public void addInternalTransaction(AionInternalTx tx) { internalTxs.add(tx); }
 
     /**
-     * Adds a collection of internal transactions.
+     * Adds a collection of internal transactions, txs, to the internal transactions list.
      *
-     * @param txs
+     * @param txs The collection of internal transactions to add.
      */
     public void addInternalTransactions(Collection<AionInternalTx> txs) {
-        internalTxs.addAll(txs);
+        for (AionInternalTx tx : txs) {
+            if (tx != null) { this.internalTxs.add(tx); }
+        }
```

Comment:
can null AionInternalTx happens in the Collection<AionInternalTx>?

Reply:
cause I think what happens in the internal txs is once an internal tx failed, then stop the following tx executing and rollback to the previous status. That's why I think should not happen a null AionInternalTx in the collection. @yulongnuco Can you confirm my words?